components/properties/src/bidi.rs

```diff
@@ -0,0 +1,81 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::props::BidiClass;
+use icu_codepointtrie::CodePointTrie;
+use unicode_bidi::data_source::BidiDataSource;
+use unicode_bidi::BidiClass as DataSourceBidiClass;
+
+pub struct BidiClassAdapter<'a> {
+    bidi_trie: &'a CodePointTrie<'a, BidiClass>,
+}
+
+impl<'a> BidiClassAdapter<'a> {
+    pub fn new(bidi_trie: &'a CodePointTrie<'a, BidiClass>) -> BidiClassAdapter<'a> {
```

Comment:
@sffc , @Manishearth 

I got a warning that this function is never used, shall I suppress the warning?

Reply:
I have supressed it.

Reply:
You should not need to suppress the dead code warning on publicly exported code. I think this warning probably has to do with features.

Reply:
done